src/pyff/samlmd.py

```diff
@@ -841,3 +841,37 @@ def expiration(t):
             return duration2timedelta(cache_duration)
 
     return None
+
+
+def sort_entities(t, sxp=None):
+    """
+Sorts the working entities 't' by the value returned by the xpath 'sxp'
+By default, entities are sorted by 'entityID' when this method is called without 'sxp', and otherwise as
+second criteria.
+Entities where no value exists for the given 'sxp' are sorted last.
+
+:param t: An element tree containing the entities to sort
+:param sxp: xpath expression selecting the value used for sorting the entities
+"""
+    def get_key(e):
+        eid = e.attrib.get('entityID')
+        sv = None
+        try:
+            sxp_values = e.xpath(sxp, namespaces=NS, smart_strings=False)
+            try:
+                sv = sxp_values[0]
+                try:
+                    sv = sv.text
+                except AttributeError:
+                    pass
+            except IndexError:
+                log.warn("Sort pipe: unable to sort entity by '%s'. "
+                         "Entity '%s' has no such value" % (sxp, eid))
+        except TypeError:
+            pass
+
+        log.debug("Generated sort key for entityID='%s' and %s='%s'" % (eid, sxp, sv))
+        return sv is None, sv, eid
+
+    container = root(t)
+    container[:] = sorted(container, key=lambda e: get_key(e))
```

Comment:
Not strictly, but it claims to "mimic the behaviour of normal Python lists as closely as possible" [1]. 

Anyway for sorting purposes, as much as I can understand they behave exactly the same, since they are both iterables [2][3]. Or am I missing something here?

References
[1] - http://lxml.de/tutorial.html#elements-are-lists
[2] - https://docs.python.org/2/library/functions.html#sorted
[3] - http://lxml.de/api.html#iteration

Reply:
I guess that is the sort of analysis I was looking for. Makes sense as a first approxmition at least!